Lib/slapdtest/_slapdtest.py

```diff
@@ -425,13 +459,18 @@ def _start_slapd(self):
                 self._log.debug(
                     "slapd connection check to %s", self.default_ldap_uri
                 )
-                self.ldapwhoami()
+                # try all bound sockets to make sure the daemon is ready to go.
+                self.ldapwhoami(ldap_uri=self.ldap_uri)
+                self.ldapwhoami(ldap_uri=self.ldaps_uri)
+                if self.ldapi_uri:
+                    self.ldapwhoami(ldap_uri=self.ldapi_uri)
```

Comment:
If one uri is contactable, the others will too, no need to check all.

Reply:
I was actually seeing sporadic failures on some GHA runs which appear to have stopped with this addition.  The specific failure from `pypy3` with a `Bad file descriptor` during `tearDown()` after `Test03_SimpleLDAPObjectWithFileno.test_attrlist_accepts_unicode`.  You might be able to [see this log](https://github.com/mdavidsaver/python-ldap/runs/5470738478?check_suite_focus=true#step:7:77).

I didn't dig into the specifics of this test failure as I'm already digressing from my project, and I had an intuition that incorrect startup was involved.  Maybe the test is swallowing a `connect()` failure?

Consider that the `slapd` process is not instantaneously/atomically calling `bind()` and `accept()` on all sockets.  So there will be a period of time, however short, when some sockets can be `connect()`ed, and others can't.  I know this seems unlikely, but I've found that CI has a way of triggering these sorts of races.